// @shoelace-style/shoelace/dist/chunks/chunk.JJKKOZIV.js
import {
  SlSelect
} from "./chunk.GTUXXKTG.js";
import {
  __objRest
} from "./chunk.5F6CPPXO.js";

// src/react/select/index.ts
import * as React from "react";
import { createComponent } from "@lit-labs/react";
var tagName = "sl-select";
var component = createComponent({
  tagName,
  elementClass: SlSelect,
  react: React,
  events: {
    onSlChange: "sl-change",
    onSlClear: "sl-clear",
    onSlInput: "sl-input",
    onSlFocus: "sl-focus",
    onSlBlur: "sl-blur",
    onSlShow: "sl-show",
    onSlAfterShow: "sl-after-show",
    onSlHide: "sl-hide",
    onSlAfterHide: "sl-after-hide",
    onSlInvalid: "sl-invalid"
  },
  displayName: "SlSelect"
});
var SlComponent = class extends React.Component {
  constructor(...args) {
    super(...args);
    SlSelect.define(tagName);
  }
  render() {
    const _a = this.props, { children } = _a, props = __objRest(_a, ["children"]);
    return React.createElement(component, props, children);
  }
};
var select_default = SlComponent;

export {
  select_default
};
